File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        first = 0
        second = 0
        merge = []
        while first < len(nums1) and second < len(nums2):
            if nums1[first] < nums2[second]:
                merge.append(nums1[first])
                first += 1
            else:
                merge.append(nums2[second])
                second += 1
        while first < len(nums1):
            merge.append(nums1[first])
            first += 1
        while second < len(nums2):
            merge.append(nums2[second])
            second += 1
        if len(merge) % 2 == 0:
            mid = len(merge) // 2
            median = (merge[mid - 1] + merge[mid]) / 2
        else:
            median = float(merge[len(merge) // 2])
        return median

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int i = 0;
        int index = 0, first = 0, second = 0;
        double median = 0.0;
        int[] merg = new int[nums1.length + nums2.length];
        while(first < nums1.length && second < nums2.length){
            if(nums1[first] < nums2[second]){
                merg[index++] = nums1[first];
                first++;
            }
            else{
                merg[index++] = nums2[second];
                second++;
            }
            i++;
        }
        while(first < nums1.length){
            merg[index++] = nums1[first++];
        }
        while(second < nums2.length){
            merg[index++] = nums2[second++];
        }
        // median = ((double)(merg[0] + merg[merg.length - 1])) / 2;
        int arrLen = nums1.length + nums2.length;
        if(arrLen % 2 == 0){
            median = ((double)(merg[arrLen/2 - 1] + merg[arrLen/2])) / 2;
            return median;
        }
        else{
            median = (double)(merg[arrLen/2]);
            return median;
        }
        
    }
}